src/scripts/clientUtils.js

```diff
@@ -0,0 +1,234 @@
+
+export function getCurrentUser() {
+    return window.ApiClient.getCurrentUser(false);
+}
+
+//TODO: investigate url prefix support for serverAddress function
+export function serverAddress() {
+    if (AppInfo.isNativeApp) {
+        const apiClient = window.ApiClient;
+
+        if (apiClient) {
+            return apiClient.serverAddress();
+        }
+
+        return null;
+    }
+
+    const urlLower = window.location.href.toLowerCase();
+    const index = urlLower.lastIndexOf('/web');
+
+    if (index != -1) {
+        return urlLower.substring(0, index);
+    }
+
+    const loc = window.location;
+    let address = loc.protocol + '//' + loc.hostname;
+
+    if (loc.port) {
+        address += ':' + loc.port;
+    }
+
+    return address;
+}
+
+export function getCurrentUserId() {
+    const apiClient = window.ApiClient;
+
+    if (apiClient) {
+        return apiClient.getCurrentUserId();
+    }
+
+    return null;
+}
+
+export function onServerChanged(userId, accessToken, apiClient) {
+    apiClient = apiClient || window.ApiClient;
+    window.ApiClient = apiClient;
+}
+
+export function logout() {
+    ConnectionManager.logout().then(function () {
+        let loginPage;
+
+        if (AppInfo.isNativeApp) {
+            loginPage = 'selectserver.html';
```

Comment:
Not sure, but we probably need to call `selectServer` instead.

Reply:
I just copied stuff out of site.js, it's not meant to fix or refactor thing, only to get code out of my way for the Webpack changes :)

But yeah, we should change this afterwards.